corpus_studies/solidity_state_study/slither_plugins/hasstate.py

```diff
@@ -0,0 +1,158 @@
+from slither.detectors.abstract_detector import AbstractDetector, DetectorClassification
```

Comment:
In the commit message, please explain what cases this analysis covers so far. It's hard to review without know what to look for.